N31: Vach (#955)
rust-gamedev/rust-gamedev.github.io: content/news/031/index.md

## Library Updates

### [vach]

[vach] is an archiving file format developed from the ground up for games and other realtime applications by [@zeskeertwee] and [@sokorototo]. It is written in pure Rust.

> It's primary objectives (in no perticular order) are:

Reviewer: ```suggestion
Its primary objectives (in no particular order) are:
```

- Have a simple, yet complete API.

> - Fine control over each individual entry in an archive, ie neighbouring entries can have vastly different compression schemes.

Reviewer: ```suggestion
- Fine control over each individual entry in an archive, i.e. neighbouring entries can have vastly different compression schemes.
```

> - Support compression (with multiple compression schemes), enryption, signatures for data authentication

Reviewer: ```suggestion
- Support compression (with multiple compression schemes), encryption, signatures for data authentication
```

> - Efficient fetching of data by avoiding unnecessary traversal of the file. Once a file is parsed locations of every entry are known to the loader.

Reviewer: ```suggestion
- Efficient fetching of data by avoiding unnecessary traversal of the file. Once a file is parsed, locations of every entry are known to the loader.
```

> - Be as compact as possible, the smallest valid archive is only 13 bytes.

Reviewer: ```suggestion
- Be as compact as possible - the smallest valid archive is only 13 bytes.
```

> - Each entry has some metadata attached to it, this is implemented using bitflags and up to 8 bits are free to the user.

Reviewer: ```suggestion
- Each entry has some metadata attached to it - this is implemented using bitflags and up to 8 bits are free to the user.
```

- Has mutlithreaded implementations of both the loader and the writer.

> A [cli][vach-cli] is allowing one to use vach as a general purpose archive format. The CLI is fully multithreaded, allowing for insane un/packing speeds. 

Reviewer: ```suggestion
A [CLI][vach-cli] is allowing one to use vach as a general purpose archive format. The CLI is fully multithreaded, allowing for insane un/packing speeds.
```

## Library Updates

### [vach]

[vach] is an archiving file format developed from the ground up for games and
other realtime applications by [@zeskeertwee] and [@sokorototo]. It is
written in pure Rust.

Its primary objectives (in no particular order) are:

- Have a simple, yet complete API.
- Fine control over each individual entry in an archive, i.e. neighbouring
  entries can have vastly different compression schemes.

> - Support compression (with multiple compression schemes), encryption,
>   signatures for data authentication

Reviewer 2: ```suggestion
- Support compression (with multiple compression schemes), encryption,
  and signatures for data authentication
```
